### PR TITLE
ENYO-4998: Add support for custom media components in VideoPlayer

### PR DESCRIPTION
**Pull request by viodragon2**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
App wants to use custom video component for a non-standard video that's not supported via HTML video element.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Add `component` prop to `VideoPlayer` to support custom components
- Add `ui/Media` component to represent a HTMLMediaElement which forwards standard HTMLMediaEvents and custom media events
- Removed forwarding media states in standard media events (currentTime, duration, etc.)

Enyo-DCO-1.1-Signed-off-by: Stephen Choi <stephen.choi@lge.com>

### CODE REVIEW

**Comment by viodragon2** [comment]
Updated per review and some more.
The documentation added to the `videoComponent` seem a bit messy. I'd like your feedback on this on what you think.

In addition to things you've pointed out, I've found that VideoPlayer had some unused/unnecessary internal states that we could get rid of. (i.e. `buffered`, `readyState`, `mute`, `volume`)

I didn't address the `customMediaEventsMap` issue you've pointed because I think I found a way to get rid of `getNode()` from `Media`. I was never a big fan of this at first, and I think it is better if we can remove it. 

Let me know what you think

**Comment by ryanjduffy** [comment]
I think this is a good start. Not sure what is a better way to handle the docs for `videoComponent` right now. I'm okay with what you have for the time being.

File: packages/ui/Media/Media.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/**
+ * Provides a represetation of HTMLMediaElement component with an ability to attach a custom event
+ * handler.
+ *
+ * @module ui/Media
+ * @exports handledMediaEventsMap
+ * @exports readyState
+ * @exports Media
```
Comment (webOS101) [inline packages/ui/Media/Media.js]: either `Media` should be `@public` or it should not appear in `@exports`.

File: packages/ui/Media/Media.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
+/**
+ * Provides a represetation of HTMLMediaElement component with an ability to attach a custom event
+ * handler.
+ *
+ * @module ui/Media
+ * @exports handledMediaEventsMap
+ * @exports readyState
```
Comment (webOS101) [inline packages/ui/Media/Media.js]: Where is `readyState` exported?

**Pull request merged**